avatar.js

```diff
@@ -214,6 +223,87 @@ function avatar_form_submit(form, form_state) {
       return;
     }
 
+    // WEBAPP
+    if (drupalgap.settings.mode === 'web-app') {
+
+      // Pull the original file name from the image upload.
+      img_file_name = form_state.values.imageURI.split('filename=')[1].split(',')[0];
+
+      avatarToDataUrl(form_state.values.imageURI, function(base64){
```

Comment:
@zak-schlemmer Instead of copying the entire `function(base64) { /* ... */ }` and pasting it here, make a new function called `avatarToDataUrlSuccess` and cut the original `function(base64) { /* ... */ }` and use that to build `avatarToDataUrlSuccess`, since the only difference between the two copy/pastes is the "filename" usage, which each caller could just pass in, for example:

```
function avatarToDataUrlSuccess(base64, filename) {

var data = {
          "file":{
            "file": base64.substring( base64.indexOf(',') + 1 ), // Remove the e.g. "data:image/jpeg;base64," from the front of the string.
            "filename": filename,
            "filepath": "public://" + filename,
            uid: Drupal.user.uid
          }
        };

     // ...... ALL THE OTHER EXISTING CODE CONTINUES HERE

}
```

Then when it comes time to use this new function you'll do something like:

```
      // Pull the original file name from the image upload.
      img_file_name = form_state.values.imageURI.split('filename=')[1].split(',')[0];

      avatarToDataUrl(form_state.values.imageURI, function(base64){
   
          avatarToDataUrlSuccess(base64, img_file_name);
     
      });
```

Now that we have that re-usable function, have the original call to avatarToDataUrl use this same technique in compiled mode. Then that whole chunk of code can be shared between the two and more easily maintained.

And on overall FYI, we'll never want to copy/paste large chunks of code like this when there is only a line or two difference between the two chunks no matter what project we're working on, we'll always abstract and make it re-use-able and more easily maintained, thanks!